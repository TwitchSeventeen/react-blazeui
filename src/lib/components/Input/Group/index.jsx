import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Group(props) {
  const {
    className, children, stacked, as: Component, rounded, roundedLeft, roundedRight, dataName, ...other
  } = props;
  const classes = classNames(
    'c-input-group',
    { 'c-input-group--stacked': stacked },
    { 'c-input-group--rounded': rounded },
    { 'c-input-group--rounded-left': roundedLeft },
    { 'c-input-group--rounded-right': roundedRight },
    className,
  );

  return (
    <Component
      {...other}
      className={classes}
      data-name={dataName}
    >
      { children }
    </Component>
  );
}

Group.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines a stacked group of Inputs
   */
  stacked: PropTypes.bool,
  /**
   * Defines a group with rounded corners
   */
  rounded: PropTypes.bool,
  /**
   * Defines a group with rounded left corners
   */
  roundedLeft: PropTypes.bool,
  /**
   * Defines a group with rounded right corners
   */
  roundedRight: PropTypes.bool,
  /**
   * Optional data-name attribute
   * (use for testing/selection)
   */
  dataName: PropTypes.string,
  /**
   * Element type to render as
   */
  as: PropTypes.elementType,
};

Group.defaultProps = {
  className: null,
  children: null,
  stacked: false,
  rounded: false,
  roundedLeft: false,
  roundedRight: false,
  dataName: 'input-group',
  as: 'div',
};

export default Group;
