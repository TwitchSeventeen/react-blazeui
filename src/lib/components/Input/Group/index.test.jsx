import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Group from './index';

describe('Field.Group', () => {
  afterEach(cleanup);

  it('should be defined', () => {
    expect(Group).toBeDefined();
  });

  it('renders correctly', () => {
    const { asFragment } = render(
      <Group>
        lorem ipsum dolor
      </Group>,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
