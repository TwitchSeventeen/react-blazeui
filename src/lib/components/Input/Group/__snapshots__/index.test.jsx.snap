// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field.Group renders correctly 1`] = `
<div
  className="c-input-group"
>
  lorem ipsum dolor
</div>
`;
